test/systemtests/swarm_test.go

```diff
@@ -0,0 +1,561 @@
+	}
+
+	logrus.Infof("Ping from %s to %s SUCCEEDED", c, ipaddr)
+	return nil
+}
+
+func (w *swarm) checkSchedulerNetworkCreated(nwName string, expectedOp bool) error {
+	logrus.Infof(w.env + "Checking whether docker network is created or not")
+	cmd := fmt.Sprintf("docker network ls | grep netplugin | grep %s | awk \"{print \\$2}\"", nwName)
+	logrus.Infof("Command to be executed is = %s", cmd)
+	op, err := w.node.runCommand(cmd)
+
+	if err == nil {
+		// if networks are NOT meant to be created. In ACI mode netctl net create should
+		// not create docker networks
+		ret := strings.Contains(op, nwName)
+		if expectedOp == false && ret != true {
+			logrus.Infof("Network names Input=%s and Output=%s are NOT matching and thats expected", nwName, op)
+		} else {
+			// If netwokrs are meant to be created. In ACI Once you create EPG,
+			// respective docker network should get created.
+			if ret == true {
+				logrus.Infof("Network names are matching.")
+				return nil
+			}
+		}
+		return nil
+	}
+	return err
+}
+
+func (w *swarm) waitForListeners() error {
+	return w.node.runCommandWithTimeOut("netstat -tlpn | grep 9090 | grep LISTEN", 500*time.Millisecond, 50*time.Second)
+}
+
+func (w *swarm) verifyVTEPs(expVTEPS map[string]bool) (string, error) {
+	var data interface{}
+	actVTEPs := make(map[string]uint32)
+
+	// read vtep information from inspect
+	cmd := "curl -s localhost:9090/inspect/driver | python -mjson.tool"
+	str, err := w.node.tbnode.RunCommandWithOutput(cmd)
+	if err != nil {
+		return "", err
+	}
+
+	err = json.Unmarshal([]byte(str), &data)
+	if err != nil {
+		logrus.Errorf("Unmarshal error: %v", err)
+		return str, err
+	}
+
+	drvInfo := data.(map[string]interface{})
+	vx, found := drvInfo["vxlan"]
+	if !found {
+		logrus.Errorf("vxlan not found in driver info")
+		return str, errors.New("vxlan not found in driver info")
+	}
+
+	vt := vx.(map[string]interface{})
+	v, found := vt["VtepTable"]
+	if !found {
+		logrus.Errorf("VtepTable not found in driver info")
+		return str, errors.New("VtepTable not found in driver info")
+	}
+
+	vteps := v.(map[string]interface{})
+	for key := range vteps {
+		actVTEPs[key] = 1
+	}
+
+	// read local ip
+	l, found := vt["LocalIp"]
+	if found {
+		switch l.(type) {
+		case string:
+			localVtep := l.(string)
+			actVTEPs[localVtep] = 1
+		}
+	}
+
+	for vtep := range expVTEPS {
+		_, found := actVTEPs[vtep]
+		if !found {
+			return str, errors.New("VTEP " + vtep + " not found")
+		}
+	}
+
+	return "", nil
+}
+func (w *swarm) verifyEPs(epList []string) (string, error) {
+	// read ep information from inspect
+	cmd := "curl -s localhost:9090/inspect/driver | python -mjson.tool"
+	str, err := w.node.tbnode.RunCommandWithOutput(cmd)
+	if err != nil {
+		return "", err
+	}
+
+	for _, ep := range epList {
+		if !strings.Contains(str, ep) {
+			return str, errors.New(ep + " not found on " + w.node.Name())
+		}
+	}
+
+	return "", nil
+}
+
+func (w *swarm) reloadNode(n *node) error {
```

Comment:
This is currently not being used because the test case skipped. This will need an abstraction for a node. Since this is very specific to vagrant. Please file an issue once PR is merged.